Remotion/Web/ClientScript/Scripts/Utilities.ts

```diff
@@ -377,44 +377,131 @@ class LayoutUtility
+  }
+
+  public static GetInnerHeight(element: Element)
+  {
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "height", "padding");
   }
 
   public static GetWidth(element: Element)
   {
-    return LayoutUtility.GetElementDimension(element, "width");
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "width");
+  }
+
+  public static GetInnerWidth(element: Element)
+  {
+    return LayoutUtility.GetElementDimensionWithElementShow(element, "width", "padding");
   }
 
-  private static GetElementDimension(element: Element, dimension: "width" | "height")
+  public static GetOuterHeight(element: HTMLElement)
+  {
+    return this.GetElementDimensionWithElementShow(element, "height", "border");
+  }
+
+  public static GetOuterWidth(element: HTMLElement)
+  {
+    return this.GetElementDimensionWithElementShow(element, "width", "border");
+  }
+
+  private static GetElementDimensionWithElementShow(element: Element, dimension: "width" | "height", extra?: "padding" | "border")
+  {
+    const htmlElement = element as HTMLElement;
+    const offsetValue = dimension === "width" ? htmlElement.offsetWidth : htmlElement.offsetHeight;
+    if (offsetValue !== 0)
+      return this.GetElementDimension(htmlElement, dimension, extra);
+
+    const previousPosition = htmlElement.style.position;
+    const previousVisibility = htmlElement.style.visibility;
+    const previousDisplay = htmlElement.style.display;
+
+    htmlElement.style.position = "absolute";
+    htmlElement.style.visibility = "hidden";
+    htmlElement.style.display = "block";
+
+    const result = this.GetElementDimension(htmlElement, dimension, extra);
+
+    htmlElement.style.position = previousPosition;
+    htmlElement.style.visibility = previousVisibility;
+    htmlElement.style.display = previousDisplay;
+
+    return result;
+  }
+
+  private static GetElementDimension(element: HTMLElement, dimension: "width" | "height", extra?: "padding" | "border")
   {
     const style = window.getComputedStyle(element);
 
-    // jQuery 1.6.4 uses the offsetWidth/offsetHeight properties to calculate element dimensions.
-    // The offsetXXX properties are rounded, while the style properties are not.
-    // We round here to make sure that we are compatible with the original behavior, but we do
-    // not use the offsetXXX properties for the calculation because they introduce unnecessary
-    // complexity for backwards compatibility that we don't need (IE).
-    const value = Math.round(parseFloat(style[dimension])) || 0;
-    
-    // For content-box: value is already correct
-    // For border-box: we need to remove padding and border from the value
-    if (style.boxSizing !== "border-box")
-      return value;
-    
     const orientations: CssOrientation[] = dimension === "width"
       ? ["Left", "Right"]
       : ["Top", "Bottom"];
 
+    const offsetValue = dimension === "width" ? element.offsetWidth : element.offsetHeight;
```

Comment:
I would like to discuss the logic of this code, in particular offsetValue>0 case and how this all works. I'm not sure if this means we need some code comments to explain the algorithm.

I'm also wondering if the offsetValue>0 and the style.boxSizing === "border-box" blocks should be unified or not. They are identical, except for the seed value but I don't know if this is by accident or by design.

Reply:
If an element is not visible its `offsetXXX` value will return `0`. If this is the case we need to calculate the dimension from the computed style since that should still provide the correct values.

I don;t understand what you mean with "offsetValue>0 and the style.boxSizing === "border-box" blocks should be unified or not"

Reply:
`offsetValue>0` please make this into a named variable and add a short reasinign comment. There's also the question now, why the invisible element doesn't need to differentiate between border-box and content-box.

unify: lines 443-461 and and 471-486 look identical to me. if we unify the `offsetValue` and `value` properties inside the relavant code paths, we could merge the two sections. then we have a signle IF branch for not-visible || border-box and and ELSE for the rest.

Question is: good idea or not (and why)